### PR TITLE
Draw win line using graphics API

### PR DESCRIPTION
**Pull request by Khufu-I**
Draw the strike through win line once a player has won. This renders the Game `winPositions` data.

Merge #38 before this.

### CODE REVIEW

File: android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt [hunk 1]
```diff
@@ -6,16 +6,45 @@ import androidx.compose.material3.Surface
...
+  fun y(row: Int) = x(Board.LENGTH - 1 - row)
+  fun toOffset(position: Int): Offset {
+    val (row, column) = position / Board.LENGTH to position % Board.LENGTH
+    return Offset(x(column), y(row))
+  }
+
+  val extension = 0.15f
+  val positions = winPositions.positions().sorted()
+  val (p1, p2) = toOffset(positions.first()) to toOffset(positions.last())
+  return lerp(p1, p2, -extension) to lerp(p1, p2, 1f + extension)
```
Comment (Mathuran) [inline android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt:27]: what is lerp? Maybe this should be aliased with a better name.

File: android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt [hunk 2]
```diff
@@ -6,16 +6,45 @@ import androidx.compose.material3.Surface
...
+}
+
 @Composable
-fun UIBoard(board: Board, enabled: Boolean, onClick: (Int) -> Unit) {
+fun UIBoard(
+  board: Board, enabled: Boolean,
+  winPositions: Bitboard? = null, onClick: (position: Int) -> Unit = {}
```
Comment (Mathuran) [inline android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt:33]: nit: formating is off here

Comment (Mathuran) [inline android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt:33]: nit: formating is off here

**Review by Mathuran — APPROVED** [review]

File: android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt (hunk 1 above)
Comment (Khufu-I) [reply android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt:27]: lerp stands for [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation) and is standard mathematical terminology used in Kotlin and other languages (i.e. C++). It's used to calculate the line between two points, in this case I'm using to extend the line between p1 and p2 by a small percentage in both directions.

File: android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt (hunk 2 above)
Comment (Khufu-I) [reply android/app/src/main/java/com/vithop/tictactoe/UIBoard.kt:33]: Can you make a code suggestion, I don't see the mis-formatting. I believe Kotlin format was applied to this code and it chose to put multiple parameters on different lines.

**Review by Vithop — APPROVED** [review]

**Pull request merged**